sklearn/tests/test_multioutput.py

```diff
@@ -864,3 +865,25 @@ def test_multioutput_regressor_has_partial_fit():
     msg = "This 'MultiOutputRegressor' has no attribute 'partial_fit'"
     with pytest.raises(AttributeError, match=msg):
         getattr(est, "partial_fit")
+
+
+@pytest.mark.parametrize(
+    "model_cls, raise_estimator, estimator",
+    [
+        (RegressorChain, LinearRegression, RandomForestRegressor),
+        (ClassifierChain, LogisticRegression, RandomForestClassifier),
+    ],
+)
+def test_chain_might_accept_nan(model_cls, raise_estimator, estimator):
```

Comment:
```suggestion
def test_chain_might_accept_non_finite(model_cls, raise_estimator, estimator):
```

Reply:
Could you split into two tests. One that check that we raise an error and one that check that we let pass through.

Please add a description to the test as well as a docstring.

Reply:
Do you mean to extend docstring of Chain classes and mention that it can handle nans?